sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -589,18 +589,20 @@ object RmmRapidsRetryIterator extends Arm with Logging {
             val splitIx = (tbl.getRowCount / 2).toInt
             withResource(tbl.contiguousSplit(splitIx)) { cts =>
               val tables = cts.map(_.getTable)
-              val batches = tables.safeMap(GpuColumnVector.from(_, spillable.dataTypes))
-              val spillables = batches.safeMap { b =>
-                SpillableColumnarBatch(
-                  b,
-                  SpillPriorities.ACTIVE_BATCHING_PRIORITY)
+              withResource(tables.safeMap(GpuColumnVector.from(_, spillable.dataTypes))) {
+                batches =>
+                  val spillables = batches.safeMap { b =>
+                    SpillableColumnarBatch(
+                      GpuColumnVector.incRefCounts(b),
```

Comment:
we shouldn't need to `incRefCounts` herel. I would consider something like this instead:
```
val batchArray = batches.toArray
val spillables = new Array[SpillableColumnarBatch](batchArray.size)
closeOnExcept(spillables) { _ => 
  closeOnExcept(batchArray) { _ => 
    batchArray.zipWithIndex.foreach { case (batch, ix) =>
      spillables[ix] = SpillableColumnarBatch(batch, SpillPriorities.ACTIVE_BATCHING_PRIORITY)
      batchArray[ix] = null
    }
  }
  require(spillables.length == 2,
                      s"Contiguous split returned ${spillables.length} tables but two were " +
                          s"expected!")
}
```

Reply:
never mind, after talking to you offline the approach you had originally is much better.